packages/modules/product/src/repositories/product.ts

```diff
@@ -69,6 +74,10 @@ export class ProductRepository extends DALUtils.mikroOrmBaseRepositoryFactory(
         }))
       }
 
+      if (isPresent(update.metadata)) {
+        update.metadata = mergeMetadata(product.metadata ?? {}, update.metadata)
+      }
```

Comment:
Q: could you try `mergeObjectProperties` in the assign bellow? This is what we are using in the base repository

Reply:
This doesn't delete a key if it is set to "" like we do in `mergeMetadata`, right?